### PR TITLE
Add CancellationSupport to ITIcketStore

### LINKED ISSUE #32443: ITicketStore async methods should support cancellation.
https://github.com/dotnet/aspnetcore/blob/4d3191176696d69a289ec465c38f2374553ed324/src/Security/Authentication/Cookies/src/ITicketStore.cs#L14-L41

### Describe the bug

These methods should accept a cancellation token. Other async interefaces support cancellation, for example: https://github.com/dotnet/aspnetcore/blob/4d3191176696d69a289ec465c38f2374553ed324/src/Identity/Extensions.Core/src/IUserStore.cs#L14-L22

### PR DESCRIPTION
**Pull request by stefannikolei**
Fixes #32443

### CODE REVIEW

**Comment by Tratcher** [comment]
Thanks for picking this up.

> src/Security/Authentication/Cookies/src/ITicketStore.cs(28,22): error RS0016: (NETCORE_ENGINEERING_TELEMETRY=Build) Symbol 'StoreAsync' is not part of the declared API.

That error can be fixed by using the analyzer code-fix in visual studio. It will update this file:
https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/Cookies/src/PublicAPI.Unshipped.txt

These new API also needs to be called over in the auth handler:
https://github.com/dotnet/aspnetcore/blob/543eb190f5573273b414856d93d8c49ce93baeef/src/Security/Authentication/Cookies/src/CookieAuthenticationHandler.cs#L157
You can pass in Context.RequestAborted from there.

File: src/Security/Authentication/Cookies/src/ITicketStore.cs [hunk 1]
```diff
@@ -18,6 +19,14 @@ public interface ITicketStore
         /// <returns>The key that can be used to retrieve the identity later.</returns>
         Task<string> StoreAsync(AuthenticationTicket ticket);
 
+        /// <summary>
+        /// Store the identity ticket and return the associated key.
+        /// </summary>
+        /// <param name="ticket">The identity information to store.</param>
+        /// <param name="cancellationToken"></param>
```
Comment (ShreyasJejurkar) [inline src/Security/Authentication/Cookies/src/ITicketStore.cs:26]:
```suggestion
        /// <param name="cancellationToken">The <see cref="CancellationToken"/> used to propagate notifications that the operation should be canceled.</param>
```
You can use this comment for `cancellationToken` parameter.

**Comment by stefannikolei** [comment]
> Thanks for picking this up.
> 
> > src/Security/Authentication/Cookies/src/ITicketStore.cs(28,22): error RS0016: (NETCORE_ENGINEERING_TELEMETRY=Build) Symbol 'StoreAsync' is not part of the declared API.
> 
> That error can be fixed by using the analyzer code-fix in visual studio. It will update this file:
> https://github.com/dotnet/aspnetcore/blob/main/src/Security/Authentication/Cookies/src/PublicAPI.Unshipped.txt
> 
> These new API also needs to be called over in the auth handler:
> https://github.com/dotnet/aspnetcore/blob/543eb190f5573273b414856d93d8c49ce93baeef/src/Security/Authentication/Cookies/src/CookieAuthenticationHandler.cs#L157
> 
> 
> You can pass in Context.RequestAborted from there.
@Tratcher Done. Ty for the help

**Review by Tratcher — APPROVED** [review]

**Comment by Tratcher** [comment]
Thanks

**Pull request merged**